### ISSUE TITLE
using _GLIBCXX_DEBUG 

### ISSUE THREAD

**Issue by janmazak** [msoos/cryptominisat#545]
I would like to compile my program with `-D_GLIBCXX_DEBUG` flag, but then linking against libcryptominisat fails (it is a well-known property of this flag that it breaks binary compatibility; as far as I know, it makes the binary much slower, so cannot be used in non-debugging builds). Would it be possible to have also a debugging build with this flag included? Then I can link against it without trouble. Ideally, the resulting libcryptominisat might have a modified name (e.g. `libcryptominisat5_dbg.so`) and should be installed with "make install".

(I don't know almost anything about cmake, so have no chance to make such a change myself in any reasonably short time.)

**Comment by cipherboy** [comment]
@janmazak Sans the separate library (which you could work around with a different installation prefix `/usr/local/debug/` perhaps...) you can compile CMS with C/CXX flags to get the correct flag... Something like this might work (untested):

```bash
CXXFLAGS="-D_GLIBCXX_DEBUG" cmake -DENABLE_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX:PATH=/usr/local/debug ..
make -j 4
sudo make install
```
It'd have the added benefit of not changing library names, only paths (`-L` vs `-l`).

**Comment by msoos** [comment]
Hi,

I don't understand why

```
CXXFLAGS="-D_GLIBCXX_DEBUG" cmake -DENABLE_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX:PATH=/usr/local/debug ..
```

doesn't fit your purpose? Is there a reason why that's not OK? There are literally tens of thousands of people who have this thing installed and you request a very specific install but I don't see others needing this. Are you sure this is really needed on a large scale, effecting thousands of users? Please let me know what you think.

**Comment by janmazak** [comment]
Sorry for bothering you with this. I did not know enough of cmake in general (and details specific to cryptominisat) to make this work and hoped that perhaps the _GLIBCXX_DEBUG option could be included "out of the box".

With the suggestion above, I made some progress, but now I see that a custom build of boost libraries will be necessary to make it work. That is definitely none of cryptominisat's business, and such an option is not in high demand anyway, apparently. (I just discovered D_GLIBCXX_DEBUG today and it found some peculiar bugs in my code almost immediately, so I tried to stick it everywhere :)
